keep-ui/app/incidents/[id]/incident-activity.tsx

```diff
@@ -0,0 +1,262 @@
+  const { data: alerts, isLoading: alertsLoading } = useIncidentAlerts(
+    incident.id
+  );
+  const { data: auditEvents, isLoading: auditEventsLoading } =
+    useMultipleFingerprintsAlertAudit(alerts?.items.map((m) => m.fingerprint));
+  const {
+    data: incidentEvents,
+    isLoading: incidentEventsLoading,
+    mutate: mutateIncidentActivity,
+  } = useAlertAudit(incident.id);
+
+  const { data: users, isLoading: usersLoading } = useUsers();
+  usePollIncidentComments(incident.id);
+
+  if (
+    usersLoading ||
+    incidentEventsLoading ||
+    auditEventsLoading ||
+    alertsLoading
+  )
+    return <Loading />;
+
+  const newCommentActivity = {
+    id: "newcomment",
+    type: "newcomment",
+    timestamp: new Date().toISOString(),
+    initiator: session?.user.email,
+  };
+
+  const auditActivities =
+    auditEvents
+      ?.concat(incidentEvents || [])
+      .sort(
+        (a, b) =>
+          new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime()
+      )
+      .map((auditEvent) => {
+        const _type =
+          auditEvent.action === "A comment was added to the incident" // @tb: I wish this was INCIDENT_COMMENT and not the text..
+            ? "comment"
+            : "alert";
+        return {
+          id: auditEvent.id,
+          type: _type,
+          initiator:
+            _type === "comment"
+              ? auditEvent.user_id
+              : alerts?.items.find(
+                  (a) => a.fingerprint === auditEvent.fingerprint
+                ),
+          text: _type === "comment" ? auditEvent.description : "",
+          timestamp: auditEvent.timestamp,
+        } as IncidentActivity;
+      }) || [];
+
+  const activities = [newCommentActivity, ...auditActivities];
+
+  const chronoContent = activities?.map((activity, index) =>
+    activity.type === "newcomment" ? (
+      <IncidentActivityChronoItemComment
+        mutator={mutateIncidentActivity}
+        incident={incident}
+        key={activity.id}
+      />
+    ) : (
+      <IncidentActivityChronoItem key={activity.id} activity={activity} />
+    )
+  );
+  const chronoIcons = activities?.map((activity, index) => {
+    if (activity.type === "comment" || activity.type === "newcomment") {
+      const user = users?.find((user) => user.email === activity.initiator);
+      return (
+        <UserAvatar
+          key={`icon-${activity.id}`}
+          image={user?.picture}
+          name={user?.name ?? user?.email ?? (activity.initiator as string)}
+        />
+      );
+    } else {
+      const source = (activity.initiator as AlertDto).source[0];
+      const imagePath = `/icons/${source}-icon.png`;
+      return (
+        <Image
+          key={`icon-${activity.id}`}
+          alt={source}
+          height={24}
+          width={24}
+          title={source}
+          src={imagePath}
+        />
+      );
+    }
+  });
+
+  return (
+    <Chrono
+      items={activities?.map((activity) => ({
+        title: activity.timestamp,
```

Comment:
```suggestion
        id: activity.id,
        title: activity.timestamp,
```
Add id, as internally react-chrono uses it as react key prop. 

https://github.com/prabhuignoto/react-chrono/blob/bfd27bb013b4fe89cfcc8bf43ecb0b92c2182d83/src/components/timeline-vertical/timeline-vertical.tsx#L91